classes/Media.php

```diff
@@ -330,8 +330,9 @@ public static function getJqueryPath($version = null, $folder = null, $minifier
         }
 
         // added jQuery migrate for compatibility with new version of jQuery
-        // will be removed when using latest version of jQuery
-        $return[] = Media::getJSPath(_PS_JS_DIR_ . 'jquery/jquery-migrate-1.2.1.min.js');
+        if ($version === _PS_JQUERY_VERSION_) {
```

Comment:
Should we really need to add jquery-migrate, only when the selected version is equal to latest?

Reply:
Yes, this jquery migrate is only compatible with jquery 3, the way it works is that jquery-migrate makes sure the legacy jquery code that we wrote before will still work with jquery 3.

In fact, this method is always called using the latest version of jquery, but if for some reason in the future someone passes another version of jquery as a parameter, we don't want to call this jquery-migrate.

Reply:
Maybe we could parse the name of the version passed as a parameter and include jquery migrate if it is a version 3.x ?  It seemed overkilled but I could do it.

Reply:
If a user use `addJquery('3.2.1')` he maybe doesn't want to use jquery-migrate.
But, what if he changed the `_PS_JQUERY_VERSION_` to jQuery 2? Will it break his back office because of jquery migrate (which will be added).
Should we store the '3.4.1' elsewhere?

Reply:
This method is supposed to be generic, but it's only used for back office... so I see two ways to go about this:

- Refactor the method, maybe rename it and make it clear that this is for back office purpose, and force a version (except if we want to be able to have different versions of jquery on different bo pages?)

- Store jquery version / Jquery migrate relationships somewhere, and use it in this method.

Reply:
ping @eternoendless @matks @Quetzacoalt91  wdyt?

Reply:
Right now, `addJquery` has no effect in `FrontControllers`, `ModuleFrontControllers` and symfony controllers.

This leaves us only with `AdminControllers` (legacy BO controllers) and `ModuleAdminControllers`.

I think the `addJquery` method should be deprecated in those as well, basically because inserting a different version of jQuery will most probably break the BO.

Reply:
So you suggest to deprecate the addJquery, and to call the last version of jquery and it's migration file directly?